Assets/Scripts/StarComponent.cs

```diff
@@ -31,6 +30,7 @@ public void OnPointerDown(PointerEventData eventData)
 
     public void OnPointerEnter(PointerEventData eventData)
     {
+        initialScale = transform.localScale;

```

Comment:
what if you do pointerDown, move pointer outside and re-enter (click and drag) - would we re-set that initial scale each time?

Reply:
That path is safe because the `OnPointerExit` will get called to reset the scale.